WcaOnRails/app/views/competitions/_my_competitions_table.html.erb

```diff
@@ -7,70 +7,72 @@
     <% end %>
   <% end %>
 <% else %>
-  <table class="table competitions-table floatThead <%= past ? "table-striped" : "" %>">
-    <thead>
-      <tr>
-        <th>Name</th>
-        <th>Location</th>
-        <th>Date</th>
-        <th></th>
-        <th></th>
-        <th></th>
-        <th class="big-column"></th>
-      </tr>
-    </thead>
-    <tbody>
-      <% competitions.each do |competition| %>
-        <% message = "" %>
-        <% if registration = competition.registrations.find_by_user_id(current_user.id) %>
-          <% if registration.accepted? %>
-            <% message = "You are registered. " %>
-          <% else %>
-            <% message = "You are currently on the waiting list. " %>
-          <% end %>
-        <% end %>
-        <% if competition.isConfirmed %>
-          <% if competition.showAtAll %>
-            <% message += "This competition is confirmed and visible." %>
-          <% else %>
-            <% message += "This competition is confirmed but not visible." %>
+  <div class="table-responsive">
+    <table class="table competitions-table floatThead <%= past ? "table-striped" : "" %>">
+      <thead>
+        <tr>
+          <th>Name</th>
+          <th>Location</th>
+          <th>Date</th>
+          <th></th>
+          <th></th>
+          <th></th>
+          <th class="big-column"></th>
+        </tr>
+      </thead>
+      <tbody>
+        <% competitions.each do |competition| %>
+          <% message = "" %>
+          <% if registration = competition.registrations.find_by_user_id(current_user.id) %>
+            <% if registration.accepted? %>
+              <% message = "You are registered. " %>
+            <% else %>
+              <% message = "You are currently on the waiting list. " %>
+            <% end %>
           <% end %>
-        <% else %>
-          <% if competition.showAtAll %>
-            <% message += "This competition is not confirmed but is visible." %>
+          <% if competition.isConfirmed %>
+            <% if competition.showAtAll %>
+              <% message += "This competition is confirmed and visible." %>
+            <% else %>
+              <% message += "This competition is confirmed but not visible." %>
+            <% end %>
           <% else %>
-            <% message += "This competition is not confirmed and not visible." %>
+            <% if competition.showAtAll %>
+              <% message += "This competition is not confirmed but is visible." %>
+            <% else %>
+              <% message += "This competition is not confirmed and not visible." %>
+            <% end %>
```

Comment:
I know you're just following an existing pattern, but this is (and already was) **way** too much logic for a view.

I'd move these 20 lines to a helper function, leaving something like this here:

`<% message = competition_message(competition, current_user) %>